providers/exoscale/exoscaleProvider.go

```diff
@@ -74,47 +96,69 @@ func (c *exoscaleProvider) GetZoneRecords(domain string) (models.Records, error)
 func (c *exoscaleProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
 	dc.Punycode()
 
+	domain, err := c.findDomainByName(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+
 	ctx := context.Background()
-	records, err := c.client.GetRecords(ctx, dc.Name)
+	records, err := c.client.ListDNSDomainRecords(ctx, c.apiZone, *domain.ID)
 	if err != nil {
 		return nil, err
 	}
 
 	existingRecords := make([]*models.RecordConfig, 0, len(records))
 	for _, r := range records {
-		if r.RecordType == "SOA" || r.RecordType == "NS" {
+		record, err := c.client.GetDNSDomainRecord(ctx, c.apiZone, *domain.ID, *r.ID)
+		if err != nil {
+			return nil, err
+		}
+
+		if *record.Type == "SOA" || *record.Type == "NS" {
 			continue
 		}
-		if r.Name == "" {
-			r.Name = "@"
+		if *r.Name == "" {
+			t := "@"
+			record.Name = &t
 		}
-		if r.RecordType == "CNAME" || r.RecordType == "MX" || r.RecordType == "ALIAS" || r.RecordType == "SRV" {
-			r.Content += "."
+		if *record.Type == "CNAME" || *record.Type == "MX" || *record.Type == "ALIAS" || *record.Type == "SRV" {
+			t := *record.Content + "."
+			// for SRV records we need to aditionally prefix target with priority, which API handles as separate field.
+			if *record.Type == "SRV" && record.Priority != nil {
+				t = fmt.Sprintf("%s %s", strconv.FormatInt(*record.Priority, 10), t)
+			}
+			record.Content = &t
 		}
 		// exoscale adds these odd txt records that mirror the alias records.
 		// they seem to manage them on deletes and things, so we'll just pretend they don't exist
-		if r.RecordType == "TXT" && strings.HasPrefix(r.Content, "ALIAS for ") {
+		if *record.Type == "TXT" && strings.HasPrefix(*record.Content, "ALIAS for ") {
 			continue
 		}
-		rec := &models.RecordConfig{
-			TTL:      uint32(r.TTL),
-			Original: r,
+
+		rc := &models.RecordConfig{
+			TTL:      uint32(*record.TTL),
+			Original: record,
 		}
-		rec.SetLabel(r.Name, dc.Name)
-		switch rtype := r.RecordType; rtype {
+		rc.SetLabel(*record.Name, dc.Name)
+
+		switch rtype := *record.Type; rtype {
 		case "ALIAS", "URL":
-			rec.Type = r.RecordType
-			rec.SetTarget(r.Content)
+			rc.Type = *record.Type
+			rc.SetTarget(*record.Content)
```

Comment:
The code `*record.Content` is repeated many times.  Maybe set a variable to the value and use the variable?